components/scream/src/share/atm_process/atmosphere_process_group.cpp

```diff
@@ -190,6 +190,57 @@ gather_internal_fields  () {
   }
 }
 
+void AtmosphereProcessGroup::
+add_conservation_checks (const std::shared_ptr<MassAndEnergyConservationCheck>& conservation_check) const
+{
+  // Loop over atm processes and add mass and energy checker where relevant
+  for (auto atm_proc : m_atm_processes) {
+
+    // We only want the check on an idividual processes. If
+    // atm_proc is a group, recursively call this function.
+    auto atm_proc_group = std::dynamic_pointer_cast<AtmosphereProcessGroup>(atm_proc);
+    if (atm_proc_group) {
+      atm_proc_group->add_conservation_checks(conservation_check);
+      continue;
+    }
+
+    // Since the checker is column-wise, only run if atm process is a Physics process.
+    if (atm_proc->type() != AtmosphereProcessType::Physics) continue;
+ 
+    // Query the computed fields for this atm process and see if either the mass or energy computation
+    // might be changed after the process has run.
+    const std::string phys_grid_name  = conservation_check->get_grid()->name();
+    const bool updates_static_energy  = atm_proc->has_computed_field("T_mid", phys_grid_name);
+    const bool updates_kinetic_energy = atm_proc->has_computed_field("horiz_winds", phys_grid_name);
+    const bool updates_water_vapor    = atm_proc->has_computed_field("qv", phys_grid_name);
+    const bool updates_water_liquid   = atm_proc->has_computed_field("qc", phys_grid_name) ||
+                                        atm_proc->has_computed_field("qr", phys_grid_name);
+    const bool updates_water_ice      = atm_proc->has_computed_field("qi", phys_grid_name);
+    const bool mass_or_energy_is_updated = updates_static_energy || updates_kinetic_energy ||
+                                           updates_water_vapor   || updates_water_liquid ||
+                                           updates_water_ice;
+
+    // If any field used in the mass or energy calculate is updated by this process,
+    // add postcondition check.
+    // Assert that, if a process adds the postcondition check, it also defines
+    // the boundary fluxes.
+    if (mass_or_energy_is_updated) {
+      const bool has_all_boundary_fluxes = atm_proc->has_computed_field("vapor_flux", phys_grid_name) &&
```

Comment:
Why do you require all physics parametrizations to compute all these fields? I saw some fluxes are set to 0 in SHOC. Is this to avoid having the property check do a bunch of if/else based on whether fields were just updated?

Reply:
The issue is that if one process includes it, it exists in the checker for all processes. So the only way to be sure it doesn't interfere is to set it to 0. I could zero all of them out before a process runs, and then have the process set the ones it needs.